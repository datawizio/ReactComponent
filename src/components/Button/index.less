@import "~antd/es/button/style/index";
@import "../../styles/variables";

.dw-btn.ant-btn {
  &.no-border {
    border: none;
    box-shadow: none;
  }

  &.highlight:not(:hover) {
    background: none;
    border: none;
    box-shadow: none;
  }

  &.highlight:hover {
    .btn-default;
  }

  &-link {
    padding-left: 0;
    padding-right: 0;
  }

  &.ant-btn-primary[disabled] {
    .btn-primary;

    &:before {
      display: block;
    }
  }
}

.theme-dark .dw-btn.ant-btn.ant-btn-primary[disabled][disabled] {
  color: gray !important;
}
