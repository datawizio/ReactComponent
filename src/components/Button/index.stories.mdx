import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import { Button } from "./index";

<Meta title="Components/Button" component={Button} />

# Button

Компонент `Button`. Базовый компонент:
[**antd/button**](https://ant.design/components/button/).

## Overview

<Preview>
  <div className="flex-container padding-between">
    <Button type="primary">Primary</Button>
    <Button type="primary" ghost>
      Ghost
    </Button>
    <Button type="default">Default</Button>
    <Button type="dashed">Dashed</Button>
    <Button type="link">Link</Button>
  </div>
</Preview>

### Dangers

<Preview>
  <div className="flex-container padding-between">
    <Button type="primary" danger>
      Primary
    </Button>
    <Button type="primary" ghost danger>
      Ghost
    </Button>
    <Button type="default" danger>
      Default
    </Button>
    <Button type="dashed" danger>
      Dashed
    </Button>
    <Button type="link" danger>
      Link
    </Button>
  </div>
</Preview>

## Рrimary

<Preview>
  <Story name="Primary">
    <Button type="primary">Primary</Button>
  </Story>
</Preview>

### Disabled

<Preview>
  <Button type="primary" disabled>
    Primary
  </Button>
</Preview>

### Loading

<Preview>
  <Button type="primary" loading>
    Primary
  </Button>
</Preview>

### Block

<Preview>
  <Button type="primary" block>
    Primary
  </Button>
</Preview>

## Default button

<Preview>
  <Story name="Default">
    <Button type="default">Default</Button>
  </Story>
</Preview>

### Disabled

<Preview>
  <Button type="default" disabled>
    Default
  </Button>
</Preview>

### Loading

<Preview>
  <Button type="default" loading>
    Default
  </Button>
</Preview>

## Dashed button

<Preview>
  <Story name="Dashed">
    <Button type="dashed">Dashed</Button>
  </Story>
</Preview>

### Disabled

<Preview>
  <Button type="dashed" disabled>
    Dashed
  </Button>
</Preview>

### Loading

<Preview>
  <Button type="dashed" loading>
    Dashed
  </Button>
</Preview>

## Link button

<Preview>
  <Story name="Link">
    <Button type="link">Link</Button>
  </Story>
</Preview>

### Disabled

<Preview>
  <Button type="link" disabled>
    Link
  </Button>
</Preview>

### Loading

<Preview>
  <Button type="link" loading>
    Link
  </Button>
</Preview>

## Props

- [**antd/button**](https://ant.design/components/button/#API)

<Props of={Button} />
