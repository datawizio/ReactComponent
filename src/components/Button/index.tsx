import React from "react";

import { Button as AntButton } from "antd";
import { ButtonProps as AntButtonProps } from "antd/lib/button";

import clsx from "clsx";

import "./index.less";

export interface ButtonProps extends AntButtonProps {
  /** Display a border around the button */
  border?: boolean;

  /** Highlight the button */
  highlight?: boolean;
}

const Button: React.FC<ButtonProps> = ({
  border,
  highlight,
  className,
  ...props
}) => {
  return (
    <AntButton
      {...props}
      className={clsx("dw-btn", className, {
        "no-border": !border,
        "highlight": highlight
      })}
    />
  );
};

Button.defaultProps = {
  border: true
};

export default Button;
