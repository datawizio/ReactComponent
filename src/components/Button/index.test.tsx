import "jsdom-global/register";
import "@testing-library/jest-dom";

import React from "react";
import { render, screen } from "@testing-library/react";

import Button from "./index";

const mockProps = {
  border: false,
  className: "custom-class-name"
};

describe("Button component", () => {
  it("is rendered correctly and matches snapshot", () => {
    const { container } = render(<Button {...mockProps}>Click me</Button>);
    expect(container).toMatchSnapshot();
  });

  it("has the correct base class", () => {
    render(<Button {...mockProps}>Click me</Button>);
    const button = screen.getByRole("button");
    expect(button).toHaveClass("dw-btn");
  });

  it("is rendered without border and applies the correct class", () => {
    render(<Button {...mockProps}>Click me</Button>);
    const button = screen.getByRole("button");
    expect(button).toHaveClass("no-border");
  });

  it("is rendered with border and does not have the no-border class", () => {
    render(
      <Button {...mockProps} border={true}>
        Click me
      </Button>
    );
    const button = screen.getByRole("button");
    expect(button).not.toHaveClass("no-border");
  });

  it("applies the correct custom class name", () => {
    render(<Button {...mockProps}>Click me</Button>);
    const button = screen.getByRole("button");
    expect(button).toHaveClass(mockProps.className);
  });

  it("renders button text correctly", () => {
    render(<Button {...mockProps}>Click me</Button>);
    expect(screen.getByText("Click me")).toBeInTheDocument();
  });
});
