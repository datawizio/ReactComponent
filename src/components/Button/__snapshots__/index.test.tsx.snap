// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Render button correctly 1`] = `
<Button
  border={false}
  className="customClassName"
>
  <Button
    className="dw-btn customClassName no-border"
  >
    <Wave>
      <button
        className="ant-btn dw-btn customClassName no-border"
        onClick={[Function]}
        type="button"
      >
        <LoadingIcon
          existIcon={false}
          loading={false}
          prefixCls="ant-btn"
        >
          <CSSMotion
            motionName="ant-btn-loading-icon-motion"
            onAppearActive={[Function]}
            onAppearStart={[Function]}
            onEnterActive={[Function]}
            onEnterStart={[Function]}
            onLeaveActive={[Function]}
            onLeaveStart={[Function]}
            removeOnLeave={true}
            visible={false}
          >
            <DomWrapper />
          </CSSMotion>
        </LoadingIcon>
      </button>
    </Wave>
  </Button>
</Button>
`;
